fetcher/fedora.go

```diff
@@ -0,0 +1,355 @@
+		}
+		results.merge(&moduleResults)
+	}
+
+	return results, nil
+}
+
+func newFedoraFetchRequests(target []string) (reqs []fetchRequest, moduleReqs [][]fetchRequest) {
+	const href = "https://dl.fedoraproject.org/pub/fedora/linux/updates/%s/Everything/x86_64/repodata/repomd.xml"
+	const moduleHref = "https://dl.fedoraproject.org/pub/fedora/linux/updates/%s/Modular/%s/repodata/repomd.xml"
+	moduleArches := []string{"x86_64", "aarch64"}
+	for _, v := range target {
+		reqs = append(reqs, fetchRequest{
+			target:       v,
+			url:          fmt.Sprintf(href, v),
+			mimeType:     mimeTypeXML,
+			concurrently: true,
+		})
+	}
+	for i, arch := range moduleArches {
+		moduleReqs = append(moduleReqs, []fetchRequest{})
+		for _, v := range target {
+			moduleReqs[i] = append(moduleReqs[i], fetchRequest{
+				target:       v,
+				url:          fmt.Sprintf(moduleHref, v, arch),
+				mimeType:     mimeTypeXML,
+				concurrently: true,
+			})
+		}
+	}
+	return
+}
+
+func fetchEverythingFedora(reqs []fetchRequest) (FedoraUpdatesPerVersion, error) {
+	log15.Info("start fetch data from repomd.xml of non-modular package")
+	feeds, err := fetchFeedFilesFedora(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch feed file, err: %w", err)
+	}
+
+	updates, err := fetchUpdateInfosFedora(feeds)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch updateinfo, err: %w", err)
+	}
+
+	results, err := parseFetchResultsFedora(updates)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to parse fetch results, err: %w", err)
+	}
+
+	for version, v := range results {
+		log15.Info(fmt.Sprintf("%d CVEs for Fedora %s Fetched", len(v.UpdateList), version))
+	}
+
+	return results, nil
+}
+
+func fetchModulesFedora(reqs []fetchRequest) (FedoraUpdatesPerVersion, error) {
+	log15.Info("start fetch data from repomd.xml of modular")
+	feeds, err := fetchModuleFeedFilesFedora(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch feed file, err: %w", err)
+	}
+
+	updates, err := fetchUpdateInfosFedora(feeds)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch updateinfo, err: %w", err)
+	}
+
+	moduleYaml, err := fetchModulesYamlFedora(feeds)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch module info, err: %w", err)
+	}
+
+	results, err := parseFetchResultsFedora(updates)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to parse fetch results, err: %w", err)
+	}
+
+	for version, result := range results {
+		log15.Info(fmt.Sprintf("%d CVEs for Fedora %s modules Fetched", len(result.UpdateList), version))
+		for i, update := range result.UpdateList {
+			yml, ok := moduleYaml[version][update.Title]
+			if ok {
+				var pkgs []Package
+				for _, rpm := range yml.Data.Artifacts.Rpms {
+					pkg, err := rpm.NewPackageFromRpm()
+					if err != nil {
+						return nil, xerrors.Errorf("Failed to build package info from rpm name, err: %w", err)
+					}
+					pkgs = append(pkgs, pkg)
+				}
```

Comment:
I'd like to make the nest shallower.
```suggestion
			if !ok {
				continue
			}
			var pkgs []Package
			for _, rpm := range yml.Data.Artifacts.Rpms {
				pkg, err := rpm.NewPackageFromRpm()
				if err != nil {
					return nil, xerrors.Errorf("Failed to build package info from rpm name, err: %w", err)
				}
				pkgs = append(pkgs, pkg)
```